src/bip21.rs

```diff
@@ -0,0 +1,102 @@
+use percent_encoding::{utf8_percent_encode, AsciiSet, CONTROLS};
+use std::collections::BTreeMap;
+use url::{ParseError, Url};
+
+pub struct Bip21 {
+    pub scheme: String,
+    pub address: String,
+    pub amount: Option<String>,
```

Comment:
did you consider using `bitcoin::Address` and `bitcoin::Amount` here?

Reply:
+1